packages/suite/src/actions/wallet/moveLabelsForRbf.ts

```diff
@@ -0,0 +1,154 @@
+    accountOutputLabels,
+    dispatch,
+    accountKey,
+    newTxid,
+}: MoveLabelToNewTransactionParams) => {
+    // eslint-disable-next-line no-restricted-syntax,
+    for (const outputIndex of Object.keys(accountOutputLabels)) {
+        const value = accountOutputLabels[outputIndex];
+        // eslint-disable-next-line no-await-in-loop
+        await dispatch(
+            metadataActions.addMetadata({
+                type: 'outputLabel',
+                entityKey: accountKey,
+                txid: newTxid,
+                outputIndex: Number(outputIndex),
+                defaultValue: '',
+                value,
+            }),
+        );
+    }
+};
+
+type FindLabelsToBeMovedOrDeletedParams = {
+    prevTxid: string;
+    getState: GetState;
+};
+
+export type LabelsToBeMovedOrDeleted = Record<
+    AccountKey,
+    {
+        toBeMoved: WalletAccountTransaction;
+        toBeDeleted: WalletAccountTransaction[];
+    }
+>;
+
+export const findLabelsToBeMovedOrDeleted = ({
+    prevTxid,
+    getState,
+}: FindLabelsToBeMovedOrDeletedParams): LabelsToBeMovedOrDeleted => {
+    const result: LabelsToBeMovedOrDeleted = {};
+
+    const accountTransactions = findTransactions(
+        prevTxid,
+        getState().wallet.transactions.transactions,
+    );
+
+    // eslint-disable-next-line no-restricted-syntax
+    for (const accountTransaction of accountTransactions) {
+        const chainedTransactionsToDrop = findChainedTransactions(
+            accountTransaction.tx.descriptor,
+            accountTransaction.tx.txid,
+            getState().wallet.transactions.transactions,
+        );
+
+        const allAccountsTransactionsIncludingChained: WalletAccountTransaction[] = [
+            accountTransaction.tx,
+            ...(chainedTransactionsToDrop?.own ?? []),
+            // Intentionally using `chainedTransactionsToDrop?.others`, they will be found when we query another account in the loop
+        ];
+
+        result[accountTransaction.key] = {
+            toBeDeleted: allAccountsTransactionsIncludingChained,
+            toBeMoved: accountTransaction.tx,
+        };
+    }
+
+    return result;
+};
+
+type MoveLabelsForRbfParams = {
+    newTxid: string;
+    toBeMovedOrDeletedList: LabelsToBeMovedOrDeleted;
+};
+
+export const moveLabelsForRbf =
+    ({ toBeMovedOrDeletedList, newTxid }: MoveLabelsForRbfParams) =>
+    async (dispatch: Dispatch, getState: GetState) => {
+        // eslint-disable-next-line no-restricted-syntax
+        for (const toBeMovedOrDeleted of Object.entries(toBeMovedOrDeletedList)) {
+            const [accountKey, data] = toBeMovedOrDeleted;
+
+            const accountMetadata = selectLabelingDataForAccount(getState(), accountKey);
+            const accountOutputLabelsToBeMoved: AccountOutputLabels =
+                accountMetadata?.outputLabels?.[data.toBeMoved.txid] ?? {};
+
+            // eslint-disable-next-line no-await-in-loop
+            await moveLabelToNewTransaction({
```

Comment:
nit: consider renaming the action. Move sounds like the old ones are deleted as a part of this action